### PR TITLE
Refs [TASK] [34080] [Design basic structure ]

### PR DESCRIPTION
**Pull request by Hoang-Kelvin**
## Related Tickets
- [#Task 34080: Design basic structure]({https://edu-redmine.sun-asterisk.vn/issues/xxxx})
## WHAT
- Create base classes for project
- Add some libraries needed for project
- Create splash screen
- Create bottom navigation
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 <resources>
-    <string name="app_name">Gamer</string>
+    <string name="app_name">GameDB</string>
     <!-- TODO: Remove or change this placeholder text -->
-
+    <string name="Saved">Saved</string>
```
Comment (thonc-2135) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="string_saved">Saved</string>
``` 
or 
```suggestion
    <string name="title _saved">Saved</string>
```

File: app/src/main/res/values-night/themes.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
         <item name="colorSecondaryVariant">@color/color_teal_200</item>
         <item name="colorOnSecondary">@android:color/black</item>
         <!-- Status bar color. -->
-        <item name="android:statusBarColor">?attr/colorPrimaryVariant</item>
+        <item name="android:statusBarColor" tools:targetApi="l">?attr/colorPrimaryVariant</item>
```
Comment (thonc-2135) [inline app/src/main/res/values-night/themes.xml]: sao target api lại là 1 v a

File: app/src/main/res/layout/activity_splash.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.splash.SplashActivity">
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (thonc-2135) [inline app/src/main/res/layout/activity_splash.xml]:
```suggestion
    tools:context=".ui.splash.SplashActivity"/>

```

File: app/src/main/java/com/project/gamedb/base/BaseAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    private val items = mutableListOf<T>()
+
+    override fun onBindViewHolder(holder: BaseViewHolder<T>, position: Int) {
+        holder.onBind(items[position])
+    }
+
+    override fun getItemId(position: Int): Long {
+        return position.toLong()
+    }
```
Comment (thonc-2135) [inline app/src/main/java/com/project/gamedb/base/BaseAdapter.kt]:
```suggestion
    override fun getItemId(position: Int) = position.toLong()
    }
```

File: app/src/main/java/com/project/gamedb/base/BaseAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    override fun getItemViewType(position: Int): Int {
+        return position
+    }
```
Comment (thonc-2135) [inline app/src/main/java/com/project/gamedb/base/BaseAdapter.kt]: tương tự như trên

File: app/src/main/res/values-night/themes.xml (hunk 2 above)
Comment (Hoang-Kelvin) [reply app/src/main/res/values-night/themes.xml]: Thực ra phải là 21

**Review by thonc-2135 — APPROVED** [review]

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**